python/shark_turbine/importers/onnx_importer.py

```diff
@@ -88,7 +88,17 @@ class GraphInfo:
     def __init__(self, model_info: ModelInfo, graph_proto: onnx.GraphProto):
         self.model_info = model_info
         self.graph_proto = graph_proto
-        self.node_map: dict[str, onnx.NodeProto] = {n.name: n for n in graph_proto.node}
+        self.node_map: dict[str, onnx.NodeProto] = {}
```

Comment:
The type annotation is no longer correct.

Let's not make this polymorphic. Just make it be a map from str to list of nodes always. Provide and use a helper method like `get_single_named_node` for existing access and assert that there is only one.